admin/class-h5p-plugin-admin.php

```diff
@@ -159,6 +162,54 @@ function add_settings_link($links) {
     return $links;
   }
 
+
+  public function ajax_get_user_h5p_save_data() {
+    global $wpdb;
+    $plugin = H5P_Plugin::get_instance();
+    $core = $plugin->get_h5p_instance('core');
+    $action = $_GET['action'];
+    $content_id = $_GET['content_id'];
+    
+
+    if(!isset($action)) {
+      $core->h5pF->setErrorMessage(__('No action set', $this->plugin_slug));
+      return;
+    }
+    
+    if(strcmp($action, 'get_content_save_data') !== 0) {
+      $core->h5pF->setErrorMessage(__('Invalid action set', $this->plugin_slug));
+      return;
+    }
+
+    $user_id = get_current_user_id();
+
+    if(!isset($user_id) || !is_numeric($user_id))
+    {
+        error_log("Could not retrieve current user");
+        return;
+    }
+
+    try{
+      $save_data = $wpdb->get_var($wpdb->prepare(
+        "SELECT `data`
+        FROM {$wpdb->prefix}h5p_contents_user_data
+        WHERE user_id = %d
+        AND content_id = %d",
+        $user_id,
+        $content_id
+      ));
+
+      wp_send_json($save_data);
+    }
+    catch(Exception $ex) {
+      error_log("Error Message: " . $ex->getMessage());
+      return;
```

Comment:
Are you swallowing the error? If something is failing the frontend should get a 500